otlp/common_test.go

```diff
@@ -52,48 +50,45 @@ func TestParseHttpHeadersIntoRequestInfo(t *testing.T) {
 
 func TestAddAttributesToMap(t *testing.T) {
 	testCases := []struct {
-		key       string
 		expected  interface{}
 		attribute *common.KeyValue
 	}{
 		{
-			key:      "str-attr",
-			expected: "str-value",
+			expected: map[string]interface{}{"str-attr": "str-value"},
 			attribute: &common.KeyValue{
 				Key: "str-attr", Value: &common.AnyValue{Value: &common.AnyValue_StringValue{StringValue: "str-value"}},
 			},
 		},
 		{
-			key:      "int-attr",
-			expected: int64(123),
+			expected: map[string]interface{}{"int-attr": int64(123)},
 			attribute: &common.KeyValue{
 				Key: "int-attr", Value: &common.AnyValue{Value: &common.AnyValue_IntValue{IntValue: 123}},
 			},
 		},
 		{
-			key:      "double-attr",
-			expected: float64(12.3),
+			expected: map[string]interface{}{"double-attr": float64(12.3)},
 			attribute: &common.KeyValue{
 				Key: "double-attr", Value: &common.AnyValue{Value: &common.AnyValue_DoubleValue{DoubleValue: 12.3}},
 			},
 		},
 		{
-			key:      "bool-attr",
-			expected: true,
+			expected: map[string]interface{}{"bool-attr": true},
 			attribute: &common.KeyValue{
 				Key: "bool-attr", Value: &common.AnyValue{Value: &common.AnyValue_BoolValue{BoolValue: true}},
 			},
 		},
 		{
-			key:      "empty-key",
-			expected: nil,
+			expected: map[string]interface{}{},
 			attribute: &common.KeyValue{
 				Key: "", Value: &common.AnyValue{Value: &common.AnyValue_StringValue{StringValue: "str-value"}},
 			},
 		},
 		{
-			key:      "array-attr",
-			expected: "[\"one\",true,3]\n",
+			expected: map[string]interface{}{
```

Comment:
I know this is an old test, but I've just realized OTel requires that arrays be homogenous types, so this is a test of something that can never happen.

Reply:
Yeah, it shouldn't happen but also serves as a secondary test of different attribute values. No harm in keeping it 🤷🏻‍♂️